http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpWsmDao.scala

```diff
@@ -0,0 +1,145 @@
+    httpClient.expectOr[CreateNetworkResponse](
+      Request[F](
+        method = Method.POST,
+        uri = config.wsmUri
+          .withPath(
+            Uri.Path
+              .unsafeFromString(
+                s"/api/workspaces/v1/${request.workspaceId.value.toString}/resources/controlled/azure/disk"
+              )
+          ),
+        body = Stream.emits(request.asJson.noSpaces.getBytes),
+        headers = Headers(fakeAuth())
+      )
+    )(onError)
+
+  override def createVm(request: CreateVmRequest)(implicit ev: Ask[F, AppContext]): F[CreateVmResult] =
+    httpClient.expectOr[CreateVmResult](
+      Request[F](
+        method = Method.POST,
+        uri = config.wsmUri
+          .withPath(
+            Uri.Path
+              .unsafeFromString(
+                s"/api/workspaces/v1/${request.workspaceId.value.toString}/resources/controlled/azure/vm"
+              )
+          ),
+        body = Stream.emits(request.asJson.noSpaces.getBytes),
+        headers = Headers(fakeAuth())
+      )
+    )(onError)
+
+  override def getWorkspace(workspaceId: WorkspaceId)(implicit ev: Ask[F, AppContext]): F[WorkspaceDescription] =
+    httpClient.expectOr[WorkspaceDescription](
+      Request[F](
+        method = Method.GET,
+        uri = config.wsmUri
+          .withPath(
+            Uri.Path
+              .unsafeFromString(s"/api/workspaces/v1/${workspaceId.value.toString}")
+          ),
+        headers = Headers(fakeAuth())
+      )
+    )(onError)
+
+  override def deleteVm(request: DeleteVmRequest)(implicit ev: Ask[F, AppContext]): F[DeleteVmResult] =
+    httpClient.expectOr[DeleteVmResult](
+      Request[F](
+        method = Method.POST,
+        uri = config.wsmUri
+          .withPath(
+            Uri.Path
+              .unsafeFromString(
+                s"/api/workspaces/v1/${request.workspaceId.value.toString}/resources/controlled/azure/vm/${request.resourceId.id.toString}"
+              )
+          ),
+        body = Stream.emits(request.jobControl.asJson.noSpaces.getBytes),
+        headers = Headers(fakeAuth())
+      )
+    )(onError)
+
+  override def getCreateVmJobResult(request: GetJobResultRequest)(implicit ev: Ask[F, AppContext]): F[CreateVmResult] =
+    httpClient.expectOr[CreateVmResult](
+      Request[F](
+        method = Method.GET,
+        uri = config.wsmUri
+          .withPath(
+            Uri.Path
+              .unsafeFromString(
+                s"/api/workspaces/v1/${request.workspaceId.value.toString}/resources/controlled/azure/vm/create-result/${request.jobId.value}"
+              )
+          ),
+        headers = Headers(fakeAuth())
+      )
+    )(onError)
+
+  private def onError(response: Response[F])(implicit ev: Ask[F, AppContext]): F[Throwable] =
+    for {
+      context <- ev.ask
+      body <- response.bodyText.compile.foldMonoid
+      _ <- logger.error(s"${context.traceId} | WSM call failed: $body")
```

Comment:
use `logger.error(context)("???")` so that `traceId` is consistently as a separate field